### PR TITLE
Add castable docs

### PR DESCRIPTION
**Pull request by brendt**
Docs for https://github.com/laravel/framework/pull/32129

### CODE REVIEW

File: eloquent-mutators.md [hunk 1]
```diff
@@ -411,6 +412,43 @@ Once the cast is defined, you may access the `options` attribute and it will aut
...
+By implementing the `Castable` interface, you'll need to specify the caster class in the value object.
+
+    <?php
+    
+    namespace App\ValueObjects;
+    
+    use Illuminate\Contracts\Database\Eloquent\Castable;
+    use App\Casts\Address as AddressCast;
+    
+    class Addess implements Castable
```
Comment (mbardelmeijer) [inline eloquent-mutators.md:433]: Small typo :)

File: eloquent-mutators.md [hunk 2]
```diff
@@ -411,6 +412,43 @@ Once the cast is defined, you may access the `options` attribute and it will aut
...
+    {
+        /**
+         * Get the caster class for this class.
+         *
+         * @return string
+         */
+        public static function castUsing()
+        {
+            return AddressCast;
```
Comment (mbardelmeijer) [inline eloquent-mutators.md:442]:
```suggestion
            return AddressCast::class;
```

File: eloquent-mutators.md (hunk 1 above)
Comment (brendt) [reply eloquent-mutators.md]: Solved

File: eloquent-mutators.md (hunk 2 above)
Comment (brendt) [reply eloquent-mutators.md]: Solved

**Comment by taylorotwell** [comment]
In hindsight it probably would have been nice to pass `array $arguments` to the Castable object too, so that you from within `castUsing` you could use an anonymous class...

```php
public static function castUsing(array $arguments)
{
     return new class(...$arguments) implements CastsAttributes {
        get;
        set;
     }
}

```

**Pull request merged**